src/management/token-provider.ts

```diff
@@ -47,8 +47,8 @@ export class TokenProvider {
     if (!('clientSecret' in params) && !('clientAssertionSigningKey' in params)) {
       throw new Error('Must provide a clientSecret or a clientAssertionSigningKey');
     } else if (
-      ('clientSecret' in params && params.clientSecret.length === 0) ||
-      ('clientAssertionSigningKey' in params && params.clientAssertionSigningKey.length === 0)
+      ('clientSecret' in params && params.clientSecret?.length === 0) ||
+      ('clientAssertionSigningKey' in params && params.clientAssertionSigningKey?.length === 0)
```

Comment:
Would it make sense to use `params.clientSecret!.length === 0` instead? (same for `clientAssertionSigningKey`) We know there is a clientSecret, because of the left side of the `&&`.

Reply:
Yeah, am going to take a proper look at tokenprovider in another task, I just added that change to make the playground code work

Reply:
Also, I wasn't fixing a TypeScript issue, I was fixing a runtime issue - where params has the property `clientSecret` but it's null or undefined and throws a runtime error because it doesn't have a length property. So you need the JS optional chaining operator `?` rather than the `!` TS operator.

Reply:
have reverted those changes - will take a look in the token provider task [d60887b](https://github.com/auth0/node-auth0/pull/819/commits/d60887b2acb8dfc52b883b577a1b50478962c36b)